tests/rptest/services/redpanda.py

```diff
@@ -169,6 +169,37 @@ def one_or_many(value):
         return value
 
 
+def get_cloud_storage_type(applies_only_on: list(CloudStorageType) = []):
```

Comment:
Avoid using mutable default argument types as their behaviour is surprising: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments.

You could rewrite this by defaulting to `None` and adding a check for `None` where you set `applies_only_on=[]`.

Reply:
addressed